src/lib/core/overlay/position/connected-position-strategy.ts

```diff
@@ -346,14 +348,36 @@ export class ConnectedPositionStrategy implements PositionStrategy {
     });
   }
 
-  /**
-   * Physically positions the overlay element to the given coordinate.
-   * @param element
-   * @param overlayPoint
-   */
-  private _setElementPosition(element: HTMLElement, overlayPoint: Point) {
-    element.style.left = overlayPoint.x + 'px';
-    element.style.top = overlayPoint.y + 'px';
+  /** Physically positions the overlay element to the given coordinate. */
+  private _setElementPosition(element: HTMLElement, overlayRect: ClientRect, overlayPoint: Point,
+    pos: ConnectionPositionPair) {
+
+    let viewport = this._viewportRuler.getViewportRect();
+    let yProp = pos.overlayY === 'bottom' ? 'bottom' : 'top';
+    let yValue = yProp === 'top' ? overlayPoint.y : viewport.height -
+        (overlayPoint.y + overlayRect.height);
+    let xProp: string;
+    let xValue: number;
+
+    if (this._dir === 'rtl') {
+      xProp = pos.overlayX === 'end' ? 'left' : 'right';
+    } else {
+      xProp = pos.overlayX === 'end' ? 'right' : 'left';
+    }
+
+    if (xProp === 'left') {
+      xValue = overlayPoint.x;
+    } else {
+      xValue = viewport.width - (overlayPoint.x + overlayRect.width);
+    }
+
+    ['top', 'bottom', 'left', 'right'].forEach(prop => {
+      if (prop === xProp || prop === yProp) {
+        element.style[prop] = `${prop === xProp ? xValue : yValue}px`;
+      } else {
+        element.style[prop] = null;
+      }
+    });
```

Comment:
Done with some slight tweaks to the comments to mention that the property determines which direction the element will expand to.